packages/notebook-extension/src/index.ts

```diff
@@ -347,6 +350,152 @@ export const commandEditItem: JupyterFrontEndPlugin<void> = {
   }
 };
 
+/**
+ * A plugin that provides a execution indicator item to the status bar or toolbar.
+ */
+export const executionIndicator: JupyterFrontEndPlugin<void> = {
+  id: '@jupyterlab/notebook-extension:execution-indicator',
+  autoStart: true,
+  requires: [INotebookTracker, ILabShell, ITranslator],
+  optional: [IStatusBar, ISettingRegistry],
+  activate: (
+    app: JupyterFrontEnd,
+    notebookTracker: INotebookTracker,
+    labShell: ILabShell,
+    translator: ITranslator,
+    statusBar: IStatusBar | null,
+    settingRegistry: ISettingRegistry | null
+  ) => {
+    // Set default position to toolbar.
+    let showOnToolBar = true;
+    let showProgress = true;
+    const toolbarItemName = 'executionProgress';
+
+    if (settingRegistry) {
```

Comment:
It seems that if no setting registry is provided by the application, no indicator will show up. I would refactor the code to always show the default execution indicator and only condition the configuration through settings.